test/integration/testExecute.js

```diff
@@ -163,180 +165,249 @@ describe('Execute test', function ()
+    await testUtil.connectAsync(connection);
+    await testUtil.executeCmdAsync(connection, createTableVariant);
+  });
+
+  after(async () => {
+    // Drop table
+    await testUtil.executeCmdAsync(connection, dropTableVariant);
+  });
+
+  describe('Variant XML', function () {
+    const TEST_ATTRIBUTE_NAME = "attr";
+    const TEST_ATTRIBUTE_VALUE = "attrValue";
+    const ELEMENT_VALUE_FIELD = "#text";
+    const TEST_XML_VAL = 123;
+
+    before(async () => {
+      let createFileFormatVariant = `CREATE OR REPLACE FILE FORMAT ${TEST_VARIANT_FORMAT} TYPE = XML`;
+      await testUtil.executeCmdAsync(connection, createFileFormatVariant);
+    });
+
+    beforeEach(async () => {
+      await testUtil.executeCmdAsync(connection, createStageVariant);
+    });
+
+    afterEach(async () => {
+      await testUtil.executeCmdAsync(connection, truncateTableVariant);
+      await testUtil.executeCmdAsync(connection, dropStageVariant);
+    });
+
+    const testCases =
+      [
+        {
+          name: 'xml_single_element',
+          type: 'XML',
+          fileExtension: '.xml',
+          sampleData: `<${TEST_HEADER}>${TEST_XML_VAL}</${TEST_HEADER}>`
+        },
+        {
+          name: 'xml_single_element_with_attribute',
+          type: 'XML',
+          fileExtension: '.xml',
+          sampleData: `<${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}>${TEST_XML_VAL}</${TEST_HEADER}>`,
+          ignoreAttributes: false
+        },
+        {
+          name: 'xml_with_attribute_and_custom_parser',
+          type: 'XML',
+          fileExtension: '.xml',
+          sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}>${TEST_XML_VAL}</${TEST_HEADER}></node>`,
+          ignoreAttributes: false,
+          attributeNamePrefix: '##'
+        },
+        {
+          name: 'xml_skip_attributes',
+          type: 'XML',
+          fileExtension: '.xml',
+          sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}>${TEST_XML_VAL}</${TEST_HEADER}></node>`,
+          attributeNamePrefix: '##'
+        },
+        {
+          name: 'xml_attribute_only',
+          type: 'XML',
+          fileExtension: '.xml',
+          sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}></${TEST_HEADER}></node>`,
+          ignoreAttributes: false,
+          attributeNamePrefix: '##'
+
+        },
+        {
+          name: 'xml_attribute_empty_value',
+          type: 'XML',
+          fileExtension: '.xml',
+          sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}/></node>`,
+          ignoreAttributes: false,
+          attributeNamePrefix: '##'
+        }
+      ];
+
+    const rowAsserts = (testCase, row) => {
+      // Check the column, header, and value is correct
+      const attributeWithPrefix = testCase.attributeNamePrefix + TEST_ATTRIBUTE_NAME;
+
+      if (testCase.name === 'xml_single_element') {
```

Comment:
how about rewriting these test cases to separate tests or move assert function to parameters? just to get rid of if/else code from tests

Reply:
done